datastreams/context.go

```diff
@@ -34,35 +34,19 @@ func PathwayFromContext(ctx context.Context) (p Pathway, ok bool) {
 // SetCheckpoint sets a checkpoint on the pathway found in ctx.
 // If there is no pathway in ctx, a new Pathway is returned.
 func SetCheckpoint(ctx context.Context, edgeTags ...string) (Pathway, context.Context) {
+	return SetCheckpointWithPayloadSize(ctx, 0, edgeTags...)
+}
+
+func SetCheckpointWithPayloadSize(ctx context.Context, payloadSize int64, edgeTags ...string) (Pathway, context.Context) {
 	if ctx == nil {
 		ctx = context.Background()
 	}
 	p, ok := PathwayFromContext(ctx)
 	if ok {
-		p = p.SetCheckpoint(edgeTags...)
+		p = p.SetCheckpointWithPayloadSize(payloadSize, edgeTags...)
 	} else {
-		p = NewPathway(edgeTags...)
+		p = NewPathwayWithPayloadSize(payloadSize, edgeTags...)
 	}
 	ctx = ContextWithPathway(ctx, p)
 	return p, ctx
 }
-
-// MergeContexts returns the first context which includes the pathway resulting from merging the pathways
```

Comment:
Removed as not used

Reply:
I think this is supposed to be the public API to let users merge context on their own (see the readme at https://github.com/DataDog/data-streams-go)

Reply:
True, I missed it :(